### PR TITLE
Str graphemes

### PR DESCRIPTION
**Pull request by travisstaloch**
This is mostly work done by @snprajwal in https://github.com/roc-lang/roc/pull/4286 who asked for some help with `Str.graphemes` on zulip.

discussion: https://roc.zulipchat.com/#narrow/stream/316715-contributing/topic/Missing.20Str.2Egraphemes

This pr reworks `strGraphemes()` and adds some passing zig tests.  I think more tests are needed but i'm not sure what types of tests to add. 

The tests check these 4 cases currently: 
```
1. "" -> []
2. "ab" -> ["a", "b"]
3. "éå" -> ["é", "å"]
4. "6🤔å" -> ["6", "🤔", "å"]
```

### CODE REVIEW

File: crates/compiler/builtins/bitcode/src/str.zig
Comment (snprajwal) [inline crates/compiler/builtins/bitcode/src/str.zig:1338]: I think we can remove this now that it is working

**Comment by snprajwal** [comment]
Great work @travisstaloch :rocket:

Note for GitHub: Closes #4286

File: crates/compiler/builtins/bitcode/src/str.zig
Comment (travisstaloch) [reply crates/compiler/builtins/bitcode/src/str.zig]: thanks. done.

File: crates/compiler/builtins/bitcode/src/main.zig
Comment (brian-carroll) [inline crates/compiler/builtins/bitcode/src/main.zig:148]: Could you rename this to remove the redundant prefix `str_`?
We already prefix it with `str.` inside `exportStrFn` so this is not necessary and none of the other exports do it.

Comment (brian-carroll) [reply crates/compiler/builtins/bitcode/src/main.zig]: Otherwise I don't see any issues, looks good! 👍

**Comment by brian-carroll** [comment]
That was quick! CI will fail though, bitcode.rs needs to match.

File: crates/compiler/builtins/src/bitcode.rs
Comment (brian-carroll) [inline crates/compiler/builtins/src/bitcode.rs:365]:
```suggestion
pub const STR_GRAPHEMES: &str = "roc_builtins.str.graphemes";
```

**Review by brian-carroll — DISMISSED** [review]

**Comment by brian-carroll** [comment]
It's OK I managed to do it from the PR

**Review by brian-carroll — DISMISSED** [review]

File: crates/compiler/builtins/src/bitcode.rs
Comment (travisstaloch) [reply crates/compiler/builtins/src/bitcode.rs]: thanks. already fixed this just now

**Comment by travisstaloch** [comment]
> It's OK I managed to do it from the PR

Oh neat. Didn't know you could make a commit like that. I didn't see you'd done that until after i pushed a fix. Hope I didn't mess anything up.

**Comment by brian-carroll** [comment]
Oh yeah we require signed commits. Just noticed the "blocked" message now, didn't think to check.
Here are some links about it. I won't be around for the next few hours but you can ask on Zulip.

https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits
https://superuser.com/questions/397149/can-you-gpg-sign-old-commits

**Comment by travisstaloch** [comment]
> Oh yeah we require signed commits. Just noticed the "blocked" message now, didn't think to check. Here are some links about it. I won't be around for the next few hours but you can ask on Zulip.
> 
Thanks. Wondered about that. I ran this command.  Will try to re-push and see if they're signed.
```console
git config --global commit.gpgsign true
```

**Comment by snprajwal** [comment]
It would be good to squash the latest two fixup commits with your original commit to ensure `git bisect` doesn't break. Although if @brian-carroll will be squashing the PR itself, then it shouldn't really matter

**Comment by travisstaloch** [comment]
Ok I think i figured out how to setup commit signing. Does the latest commit look ok?

**Comment by snprajwal** [comment]
Yep it is signed, but you seem to have not added your public key to GitHub. See https://docs.github.com/en/authentication/managing-commit-signature-verification/adding-a-gpg-key-to-your-github-account

**Comment by travisstaloch** [comment]
Ok i just pushed again. Not seeing 'verified' so that seems to have worked. Thanks @snprajwal for the tip.

**Comment by travisstaloch** [comment]
I also just squashed the commits. So i'm hoping that everything is signed now.

**Comment by travisstaloch** [comment]
Ok thats better. I somehow lost the last two commits during squash. Sorry for all the noise.

**Comment by travisstaloch** [comment]
One more thing i'd like to do is combine the tests for countGraphemeClusters() with the new tests i've added. I'd like to do this by adding a test helper which checks the count and the clusters, something like this:

```zig
test "graphemes: emojis, utf8, and ascii characters" {
    try testGraphemes("6🤔å🤔e¥🤔çpp", 10, .{"6", "🤔", "å", "🤔", "e", "¥", "🤔", "ç", "p", "p"});
}

```

**Review by rtfeldman — APPROVED** [review]
@travisstaloch Great work on this! I'm approving so you can merge it if you're ready; you can always open a future PR to improve test coverage, but landing it means people can start using this implementation in their projects!

**Comment by travisstaloch** [comment]
It appears that @snprajwal's commit isn't signed and that's what is blocking merging.  I attempted to preserve his commit but i'm guessing this is blocked because that was before i had signing setup properly on this machine.

Does this make sense? Any suggestions?

**Comment by snprajwal** [comment]
> Any suggestions?

- `git rebase -i HEAD~1`
- Add `exec git commit --amend --no-edit -S` after my commit
- `git push --force`

Guessing this should fix the issue

**Comment by snprajwal** [comment]
> I attempted to preserve his commit

Co-authored commits must be signed by the primary author, as a commit can be signed with only one GPG key

**Comment by travisstaloch** [comment]
looks like they are both verified now.  thanks again for the assist with signing @snprajwal!

**Review by rtfeldman — APPROVED** [review]

**Comment by rtfeldman** [comment]
Thank you both for the great work on this! 😍

**Comment by snprajwal** [comment]
Closes #4193

**Pull request merged**
